src/olympia/blocklist/utils.py

```diff
@@ -0,0 +1,144 @@
+JSON_DATE_FORMAT = "%Y-%m-%dT%H:%M:%SZ"
+
+
+def del_none(d):
+    """
+    Delete keys with the value ``None`` in a dictionary, recursively.
+
+    This alters the input so you may wish to ``copy`` the dict first.
+    """
+    # d.iteritems isn't used as you can't del or the iterator breaks.
+    for key, value in d.items():
+        if value is None:
+            del d[key]
+        elif isinstance(value, dict):
+            del_none(value)
+        elif isinstance(value, list):
+            for v in value:
+                if isinstance(v, dict):
+                    del_none(v)
+                    if not v:
+                        d[key] = []
+
+    return d  # For convenience
+
+
+def certificates_to_json(items):
+    results = []
+    for cert in items:
+        results.append({
+            'blockID': cert.block_id,
+            'serialNumber': cert.serial,
+            'issuerName': cert.issuer,
+            'details': {
+                'name': cert.details.name,
+                'who': cert.details.who,
+                'why': cert.details.why,
+                'bug': cert.details.bug,
+                'created': cert.details.created.strftime(JSON_DATE_FORMAT),
+            }
+        })
+    return [del_none(r) for r in results]
```

Comment:
In each of the `to_json` functions, instead of doing this loop calling del_none(), couldn't you call del_none() directly when appending the dict to the results ? That way `del_none()` does not need to be recursive, and you don't need to loop on the results. Thoughts ?